proposals/dao/rariTimelock.ts

```diff
@@ -0,0 +1,120 @@
+import { ethers } from 'hardhat';
+import chai, { expect } from 'chai';
+import CBN from 'chai-bn';
+import {
+  DeployUpgradeFunc,
+  MainnetContracts,
+  NamedContracts,
+  SetupUpgradeFunc,
+  TeardownUpgradeFunc,
+  ValidateUpgradeFunc
+} from '@custom-types/types';
+import { getImpersonatedSigner } from '@test/helpers';
+
+chai.use(CBN(ethers.BigNumber));
+
+/*
+Rari Exchanger Timelocks
+
+DEPLOY ACTIONS:
+
+1. Deploy QuadraticTimelockedDelegator
+2. Deploy QuadtraticTimelockedSubdelegator
+3. Deploy ExchangerTimelock x2
+
+*/
+
+const delegatorBeneficiary = '0xB8f482539F2d3Ae2C9ea6076894df36D1f632775'; // TODO change to actual Rari addresses
+const subdelegatorBeneficiary = '0xB8f482539F2d3Ae2C9ea6076894df36D1f632775'; // TODO change to actual Rari addresses
+
+const FIVE_YEARS = '157680000';
+const TIMELOCK_START = '1630000000'; // TODO set to Rari vesting start
+
+const toBN = ethers.BigNumber.from;
+
+export const deploy: DeployUpgradeFunc = async (deployAddress, addresses, logging = false) => {
+  const { pegExchanger, tribe } = addresses;
+
+  if (!pegExchanger || !tribe) {
+    throw new Error('An environment variable contract address is not set');
```

Comment:
nit: these aren't environment variables

Reply:
(environment variable has a very specific meaning so I'm going to keep saying this)

Reply:
k will fix